From: NixOS Android devices contributors <nixos-android-devices@localhost>
Date: Mon, 28 Sep 2026 20:00:00 +0800
Subject: [PATCH] arm: dts: rebase Clover on the shared upstream DT tree

Local follow-up to unmodified sdm660-mainline/u-boot commit
 a4faf7b435dcc92bc8741a98c5fb04045c7ba1f4:
https://github.com/sdm660-mainline/u-boot/commit/a4faf7b435dcc92bc8741a98c5fb04045c7ba1f4

Move the board into OF_UPSTREAM's vendor directory, use the upstream
CDSP reservation label, and omit the downstream-only fuel gauge node.
The original optional debug config is retained but never selected.
---
diff --git a/arch/arm/dts/qcom/sdm660-xiaomi-clover.dts b/dts/upstream/src/arm64/qcom/sdm660-xiaomi-clover.dts
similarity index 97%
rename from arch/arm/dts/qcom/sdm660-xiaomi-clover.dts
rename to dts/upstream/src/arm64/qcom/sdm660-xiaomi-clover.dts
--- a/arch/arm/dts/qcom/sdm660-xiaomi-clover.dts
+++ b/dts/upstream/src/arm64/qcom/sdm660-xiaomi-clover.dts
@@ -13,7 +13,7 @@
 #include <dt-bindings/pinctrl/qcom,pmic-gpio.h>
 
 /delete-node/ &adsp_region;
-/delete-node/ &buffer_mem;
+/delete-node/ &cdsp_region;
 /delete-node/ &mba_region;
 /delete-node/ &zap_shader_region;
 /delete-node/ &venus_region;
@@ -125,7 +125,7 @@
 			no-map;
 		};
 
-		buffer_mem: buffer_region@94b00000 {
+		cdsp_region: buffer_mem: buffer_region@94b00000 {
 			reg = <0x00 0x94b00000 0x00 0x600000>;
 			no-map;
 		};
@@ -191,13 +191,6 @@
 
 &pm660_charger {
 	monitored-battery = <&battery>;
-
-	status = "okay";
-};
-
-&pm660_fg {
-	monitored-battery = <&battery>;
-	power-supplies = <&pm660_charger>;
 
 	status = "okay";
 };
